test/integration/src/test/java/io/pravega/test/integration/AppendTest.java

```diff
@@ -303,6 +308,35 @@ public void appendThroughStreamingClient() throws InterruptedException, Executio
         ack.get(5, TimeUnit.SECONDS);
     }
     
+    
+    @Test(timeout = 100000)
+    public void appendALotOfData() {
```

Comment:
this test causes the `InMemoryDurableDataLog` to go OOM on my dev box...

Reply:
Looks like you're sending in 1000 x 1MB buffers for a total of 1GB. That is a bit too extreme. 

If you do want to keep this test as it is, may I recommend truncating the segment after every few writes you're sending? That should help clear memory pressure.

Reply:
Done